From aa293fe8b0eee4f21badfe93dab25528477d1b96 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Tue, 20 Dec 2022 09:42:54 +0100
Subject: [PATCH] softhddev: remove deprecated av_init_packet

Signed-off-by: Andreas Baierl <ichgeh@imkreisrum.de>
---
 softhddev.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/softhddev.c b/softhddev.c
index d15df5f..6ea4cfd 100644
--- a/softhddev.c
+++ b/softhddev.c
@@ -692,7 +692,7 @@ int PlayAudio(const uint8_t * data, int size, uint8_t id)
 			break;
 		}
 		if (r > 0) {
-			AVPacket avpkt[1];
+			AVPacket *avpkt;
 
 			// new codec id, close and open new
 			if (AudioCodecID != codec_id) {
@@ -700,12 +700,18 @@ int PlayAudio(const uint8_t * data, int size, uint8_t id)
 				CodecAudioOpen(MyAudioDecoder, codec_id, NULL, &timebase);
 				AudioCodecID = codec_id;
 			}
-			av_init_packet(avpkt);
+			avpkt = av_packet_alloc();
+			if (avpkt == NULL) {
+				Error(_("[softhddev] avpkt allocation failed\n"));
+				fprintf(stderr, "[softhddev] avpkt allocation failed\n");
+				continue;
+			};
 			avpkt->data = (void *)p;
 			avpkt->size = r;
 			avpkt->pts = AudioAvPkt->pts;
 			CodecAudioDecode(MyAudioDecoder, avpkt);
 			AudioAvPkt->pts = AV_NOPTS_VALUE;
+			av_packet_free(&avpkt);
 			p += r;
 			n -= r;
 			continue;
-- 
2.30.2

