From b565bcd27fa3a774195182614d6aa6f28fb41b41 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Mon, 12 Dec 2022 21:05:58 +0100
Subject: [PATCH] debug

---
 Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index d4b8617..1498eed 100644
--- a/Makefile
+++ b/Makefile
@@ -16,12 +16,12 @@ MMAL ?= 0
 	# Use OpenGL/ES for OSD? Disable autodetection by setting GLES=1 or GLES=0 with make command
 GLES ?= $(shell pkg-config --exists glesv2 egl gbm && echo 1)
 
-CONFIG := #-DDEBUG 				# enable debug output+functions
-#CONFIG += -DAV_SYNC_DEBUG		# enable debug messages AV_SYNC
-#CONFIG += -DSOUND_DEBUG		# enable debug messages SOUND
+CONFIG := -DDEBUG 				# enable debug output+functions
+CONFIG += -DAV_SYNC_DEBUG		# enable debug messages AV_SYNC
+CONFIG += -DSOUND_DEBUG		# enable debug messages SOUND
 #CONFIG += -DOSD_DEBUG			# enable debug messages OSD
-#CONFIG += -DDRM_DEBUG			# enable debug messages in drm configuration
-#CONFIG += -DCODEC_DEBUG		# enable debug messages in codec configuration
+CONFIG += -DDRM_DEBUG			# enable debug messages in drm configuration
+CONFIG += -DCODEC_DEBUG		# enable debug messages in codec configuration
 #CONFIG += -DSTILL_DEBUG		# still picture debug
 #CONFIG += -DMEDIA_DEBUG		# media player debug
 
-- 
2.30.2

