From f9502f46f7965674660ac8e09ab6dd5d6f9a4117 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Wed, 23 Nov 2022 21:16:06 +0100
Subject: [PATCH 4/5] move CheckZpos

---
 video_drm.c | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/video_drm.c b/video_drm.c
index d0398b2..66dd1ab 100644
--- a/video_drm.c
+++ b/video_drm.c
@@ -714,6 +714,13 @@ search_mode:
 	get_properties(render->fd_drm, render->planes[VIDEO_PLANE]->plane_id, render->planes[VIDEO_PLANE]);
 	get_properties(render->fd_drm, render->planes[OSD_PLANE]->plane_id, render->planes[OSD_PLANE]);
 
+	// Check, if we can set z-order (meson has fixed z-order, which cannot be changed)
+	if (render->use_zpos && CheckZpos(render, render->planes[VIDEO_PLANE]->plane_id, render->planes[VIDEO_PLANE]->zpos)) {
+		render->use_zpos = 0;
+	}
+	if (render->use_zpos && CheckZpos(render, render->planes[OSD_PLANE]->plane_id, render->planes[VIDEO_PLANE]->zpos)) {
+		render->use_zpos = 0;
+	}
 
 	// render->use_zpos was set, if Video is on OVERLAY, and Osd is on PRIMARY
 	// Check if the OVERLAY plane really got a higher zpos than the PRIMARY plane
@@ -2364,14 +2371,6 @@ void VideoInit(VideoRender * render)
 	if (!(ModeReq = drmModeAtomicAlloc()))
 		fprintf(stderr, "cannot allocate atomic request (%d): %m\n", errno);
 
-	// Check, if we can set z-order (meson has fixed z-order, which cannot be changed)
-	if (render->use_zpos && CheckZpos(render, render->planes[VIDEO_PLANE]->plane_id, render->planes[VIDEO_PLANE]->zpos)) {
-		render->use_zpos = 0;
-	}
-	if (render->use_zpos && CheckZpos(render, render->planes[OSD_PLANE]->plane_id, render->planes[VIDEO_PLANE]->zpos)) {
-		render->use_zpos = 0;
-	}
-
 	SetPropertyRequest(ModeReq, render->fd_drm, render->crtc_id,
 						DRM_MODE_OBJECT_CRTC, "MODE_ID", modeID);
 	SetPropertyRequest(ModeReq, render->fd_drm, render->connector_id,
-- 
2.30.2

