From cd1d2d446fc2020e478ce40941e392c7e3d33c09 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Fri, 25 Nov 2022 20:28:01 +0100
Subject: [PATCH 6/9] fix crtc width

---
 video_drm.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/video_drm.c b/video_drm.c
index fe98c94..8b190d5 100644
--- a/video_drm.c
+++ b/video_drm.c
@@ -1320,14 +1320,9 @@ page_flip:
 	uint64_t PicWidth = DispWidth;
 	uint64_t PicHeight = DispHeight;
 
-	if (frame) {
+	if (frame)
 		PicWidth = DispHeight * av_q2d(frame->sample_aspect_ratio) *
 			   frame->width / frame->height;
-		if (!PicWidth || PicWidth > DispWidth) {
-			PicWidth = DispWidth;
-			PicHeight = PicWidth * frame->height / av_q2d(frame->sample_aspect_ratio) / frame->width;
-		}
-	}
 
 	if (!PicWidth || PicWidth > DispWidth)
 		PicWidth = DispWidth;
-- 
2.30.2

