From 32d8e473fb6ae38048251b13fb686c9e4995a0c3 Mon Sep 17 00:00:00 2001
From: Klaus Schmidinger <vdr@tvdr.de>
Date: Wed, 18 Sep 2024 11:10:56 +0200
Subject: [PATCH 11/16] Fixed description of cSkinDisplayReplay::SetRecording()

---
 HISTORY | 3 ++-
 skins.h | 6 +++---
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/HISTORY b/HISTORY
index 39729efe..71ddaa40 100644
--- a/HISTORY
+++ b/HISTORY
@@ -9984,7 +9984,7 @@ Video Disk Recorder Revision History
   version numbering. Version numbers are simply counted upwards, with each of the three
   parts ("version", "major", "minor") always being a single digit, and '0' being skipped.
 
-2024-09-17:
+2024-09-18:
 
 - Fix for compilers that don't like non-constant format strings (thanks to Stefan Hofmann).
 - Deprecated code is now marked with [[deprecated]] to issue a compile time warning when
@@ -10001,3 +10001,4 @@ Video Disk Recorder Revision History
 - The number of errors in a recording now represents the number of broken frames.
 - Now distinguishing between frames with errors and completely missing frames.
 - Recording errors are now marked in the index file.
+- Fixed description of cSkinDisplayReplay::SetRecording().
diff --git a/skins.h b/skins.h
index 9fdd72da..164028a5 100644
--- a/skins.h
+++ b/skins.h
@@ -4,7 +4,7 @@
  * See the main source file 'vdr.c' for copyright information and
  * how to reach the author.
  *
- * $Id: skins.h 5.4 2024/07/15 14:42:22 kls Exp $
+ * $Id: skins.h 5.5 2024/09/18 11:06:39 kls Exp $
  */
 
 #ifndef __SKINS_H
@@ -326,8 +326,8 @@ public:
        ///< progress bar through a cProgressBar object.
   virtual void SetRecording(const cRecording *Recording);
        ///< Sets the recording that is currently being played.
-       ///< The default implementation calls SetTitle() with the title and short
-       ///< text of the Recording. A derived class can use any information provided
+       ///< The default implementation calls SetTitle() with the title
+       ///< of the Recording. A derived class can use any information provided
        ///< by the given Recording and display it.
   virtual void SetTitle(const char *Title) = 0;
        ///< Sets the title of the recording.
-- 
2.39.2

